datacollectiondemo/source/c/CMakeLists.txt

```diff
@@ -15,23 +15,31 @@
 #
 
 cmake_minimum_required(VERSION 2.8.8)
-
-project (C-SDK-sample)
-enable_language(C)
+project (C-SDK-sample LANGUAGES C)
 
 find_package (OpenSSL REQUIRED)
 
+set(APP_NAME "demo_client")
+set(KAA_SDK_DIR "client-c")
+
+execute_process(COMMAND ${CMAKE_COMMAND} -E make_directory ${KAA_SDK_DIR}
+                WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/libs/kaa"
+               )
+execute_process(COMMAND sh -c "tar -zxf kaa-c*.tar.gz -C ${KAA_SDK_DIR}"
+                WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/libs/kaa"
+               )
```

Comment:
Agree with you. Also I think it is useful to keep build of sdk and sample app in separate directories.